### ISSUE TITLE
Table field accessibility

### ISSUE THREAD

**Issue by missmatsuko** [craftcms/cms#7322]
### Description
<img width="944" alt="Screenshot of table field." src="https://user-images.githubusercontent.com/17884198/102862858-9833c200-43e6-11eb-8c36-7c5cffedff60.png">

- The label and instructions for the table might get missed by screen readers in form mode. Can attach it to the inputs, but I think that will be too verbose. Will have to experiment with this to find a solution.
- Inputs in the table cells are missing labels. Should use `aria-labelledby` to point to its column's header and a unique identifier, maybe the row number? That might get confusing with reordering though.
- Reordering and removing rows icons have insufficient contrast.
- Recording and removing rows icons should be button elements rather than anchor elements.
- Reordering and removing rows doesn't seem to be keyboard-accessible.
- Communicate min rows and max rows other than visually.

### Steps to reproduce
1. Create and inspect a table field.

### Additional info
- Craft version: 3.5.17.1

**Comment by brandonkelly** [comment]
Resolved via #10502
